src/MetaMorpho.sol

```diff
@@ -376,6 +372,10 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
             if (supplyCap == 0) revert ErrorsLib.UnauthorizedMarket(id);
 
+            if (allocation.assets == type(uint256).max) {
+                allocation.assets = totalWithdrawn - totalSupplied;
+            }
```

Comment:
I believe this is a feature that requires a dedicated discussion before being introduced in the middle of a PR like so (without being tested)